Implement the course tabs API.

jcallaway opened the pull request:
The impetus for this is that we want to hide certain tabs in new courses by default.

ToeBee commented:
You left the `public static void main` for testing in there :) I usually do this kind of testing in the `TestLauncher` class.

The visibility options *could* be turned into an enum. I haven't looked too closely yet and given how fickle the Canvas API can be with literal values that may or may not be a good idea. Thoughts?

jcallaway commented:
I removed the main method from TabImpl.  Sorry about that!

Re: creating an enum for visibility, I don't think it would really be worth it since the API currently doesn't let you change the visibility.  If they allow this, then I think that would be the time to add the enum.  I added a comment to this effect.

ToeBee commented:
Fair enough. Thanks for the contribution!
